dll/win32/shell32/utils.cpp

```diff
@@ -762,6 +762,133 @@ SHCreatePropertyBag(_In_ REFIID riid, _Out_ void **ppvObj)
     return SHCreatePropertyBagOnMemory(STGM_READWRITE, riid, ppvObj);
 }
 
+// The helper function for SHGetUnreadMailCountW
+static DWORD
+SHELL_ReadSingleUnreadMailCount(
+    _In_ HKEY hKey,
+    _Out_opt_ PDWORD pdwCount,
+    _Out_opt_ PFILETIME pFileTime,
+    _Out_writes_opt_(cchShellExecuteCommand) LPWSTR pszShellExecuteCommand,
+    _In_ INT cchShellExecuteCommand)
+{
+    DWORD dwType, dwCount, cbCount = sizeof(dwCount);
+    DWORD error = SHQueryValueExW(hKey, L"MessageCount", 0, &dwType, &dwCount, &cbCount);
+    if (error)
+        return error;
+    if (pdwCount && dwType == REG_DWORD)
```

Comment:
and `dwCount == sizeof(DWORD)`

Reply:
That check will be harder than the original code does.

Reply:
What matters is our code. What's this "original code" you're talking about, anyways?

Reply:
*Windows does.